### PR TITLE
First commit

### PR DESCRIPTION
**Pull request by loranazarenko**
Coverage1

### CODE REVIEW

File: src/main/java/mate/academy/onlinebookstore/dto/BookDto.java [hunk 1]
```diff
@@ -1,9 +1,11 @@
 package mate.academy.onlinebookstore.dto;
 
 import java.util.Set;
+import lombok.experimental.Accessors;
 
+@Accessors(chain = true)
 public record BookDto(Long id, String title, String author,
-                      String isbn, Double price, String description,
+                      String isbn, java.math.BigDecimal price, String description,
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/onlinebookstore/dto/BookDto.java]:
```suggestion
                      String isbn, BigDecimal price, String description,
```

File: src/main/java/mate/academy/onlinebookstore/dto/CategoryRequestDto.java [hunk 2]
```diff
@@ -1,7 +1,9 @@
 package mate.academy.onlinebookstore.dto;
 
 import jakarta.validation.constraints.NotEmpty;
+import lombok.experimental.Accessors;
 
+@Accessors(chain = true)
 public record CategoryRequestDto(
         @NotEmpty
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/onlinebookstore/dto/CategoryRequestDto.java]:
```suggestion
        @NotBlank
```

File: src/main/java/mate/academy/onlinebookstore/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -3,14 +3,15 @@
...
+import lombok.experimental.Accessors;
 
+@Accessors(chain = true)
 public record CreateBookRequestDto(
         @NotNull
         String title,
         @NotNull
         String author,
-        @ISBN
+        @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/onlinebookstore/dto/CreateBookRequestDto.java]:
```suggestion
        @NotBlank
```

For all String types

File: src/main/java/mate/academy/onlinebookstore/entity/Book.java [hunk 4]
```diff
@@ -23,6 +24,7 @@
 @SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
 @Where(clause = "is_deleted=false")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/onlinebookstore/entity/Book.java]: `@Where` is deprecated, use `@SQLRestriction` instead

File: src/main/java/mate/academy/onlinebookstore/entity/Category.java [hunk 5]
```diff
@@ -15,6 +16,7 @@
 @SQLDelete(sql = "UPDATE categories SET is_deleted = TRUE WHERE id = ?")
 @Where(clause = "is_deleted = FALSE")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/onlinebookstore/entity/Category.java]: the same
Check all entities

File: src/test/java/mate/academy/onlinebookstore/util/UtilsForTests.java [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+import java.util.Set;
+import java.util.stream.Collectors;
+import mate.academy.onlinebookstore.dto.BookDto;
+import mate.academy.onlinebookstore.dto.CategoryRequestDto;
+import mate.academy.onlinebookstore.dto.CategoryResponseDto;
+import mate.academy.onlinebookstore.dto.CreateBookRequestDto;
+import mate.academy.onlinebookstore.entity.Book;
+import mate.academy.onlinebookstore.entity.Category;
+
+public class UtilsForTests {
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/onlinebookstore/util/UtilsForTests.java]:
```suggestion
public class TestUtils {
```

File: src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+        assertEquals(expected.get(1), actual.get(1));
+        assertEquals(expected.get(2), actual.get(2));
+    }
+
+    @Test
+    @DisplayName("""
+        Check getBookById(). Check if return a one book by id from database
+            """)
+    @WithMockUser(roles = {"USER", "ADMIN"})
+    void getBookById_checkValidId_returnOneBook() throws Exception {
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java]:
```suggestion
    void getBookById_ValidId_Ok() throws Exception {
```

File: src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+        //Then
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("""
+        Check wrong getBookById(). Check throw a exception
+            """)
+    @WithMockUser(roles = {"USER", "ADMIN"})
+    void getBookById_checkInvalidId_returnOneBook() throws Exception {
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java]:
```suggestion
    void getBookById_InvalidId_ShouldThrowException() throws Exception {
```

File: src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+        //Given
+        Long invalidId = 9L;
+        String message = "Can't find book by id9";
+        //When Then
+        ResultActions resultActions = mockMvc.perform(get("/books/{id}", invalidId)
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java]: Or you can 
```suggestion
        mockMvc.perform(get("/books/{id}", invalidId)
                        .contentType(MediaType.APPLICATION_JSON))
                .andExpect(status().isNotFound())
                .andExpect(result -> assertTrue(result.getResolvedException()
                        instanceof EntityNotFoundException))
                .andExpect(result -> assertEquals("Can't find book by id%s"
                        .formatted(9), result.getResolvedException().getMessage()));
```

File: src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java [hunk 10]
```diff
@@ -0,0 +1,267 @@
...
+                .andExpect(jsonPath("message").value(message));
+    }
+
+    @Test
+    @DisplayName("""
+                 Check createBook(). Check if create new book, "
+                 + "add book to database and return bookDto
+                 """)
+    @WithMockUser(roles = {"ADMIN"})
+    public void createBook_checkValidData_createOneBook() throws Exception {
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java]:
```suggestion
    public void createBook_ValidRequestDto_Ok() throws Exception {
```

File: src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java [hunk 11]
```diff
@@ -0,0 +1,267 @@
...
+                BookDto.class);
+        EqualsBuilder.reflectionEquals(expected, actual, "id");
+    }
+
+    @Test
+    @DisplayName("""
+        Check updateBook(). Check if update data some book
+            """)
+    @WithMockUser(roles = {"USER", "ADMIN"})
+    void updateBook_checkValidId_updatedOneBook() throws Exception {
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java]:
```suggestion
    void updateBook_ExistingBookWithValidRequestDto_Ok() throws Exception {
```

File: src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java [hunk 12]
```diff
@@ -0,0 +1,267 @@
...
+        Check delete(). Check if delete some book
+            """)
+    @WithMockUser(roles = {"ADMIN"})
+    void delete_validId_ok() throws Exception {
+        Book deleted = UtilsForTests.createFirstBook(Set.of(UtilsForTests.createFirstCategory()));
+        mockMvc.perform(delete("/books/{id}", deleted.getId())
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+    }
+}
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java:305]: Add test to update not existing book

File: src/test/java/mate/academy/onlinebookstore/service/BookServiceTest.java [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    @Test
+    void deleteProductWithExceptionTest() {
+        doThrow(RuntimeException.class).when(bookRepository).deleteById(any());
+
+        assertThrows(RuntimeException.class,
+                () -> bookService.deleteById(UtilsForTests.SECOND_MOCK_ID));
+    }
+
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/onlinebookstore/service/BookServiceTest.java:155]: Add not valid cases

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Let's rename test methods, also add not valid cases

**Review by sarakhmen — APPROVED** [review]
good job! Let's also use more meaningful commit names next time ;)

**Pull request merged**